panic: runtime error: invalid memory address or nil pointer dereference

### Background

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x10e2072]

goroutine 39269 [running]:
github.com/lightningnetwork/lnd.(*rpcServer).extractPaymentIntent(0xc00007b860,
0xc001f4d6a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
github.com/lightningnetwork/lnd/rpcserver.go:4255 +0x3f2
github.com/lightningnetwork/lnd.(*rpcServer).sendPaymentSync(0xc00007b860, 0x176
3c80, 0xc002b48030, 0xc001f4d6a0, 0x40c4aa, 0x1451e00, 0x146dd60)
github.com/lightningnetwork/lnd/rpcserver.go:4652 +0xbc
github.com/lightningnetwork/lnd.(*rpcServer).SendPaymentSync(0xc00007b860, 0x176
3c80, 0xc002b48030, 0xc000302240, 0xc00007b860, 0xc000478940, 0x1)
github.com/lightningnetwork/lnd/rpcserver.go:4614 +0x6f
github.com/lightningnetwork/lnd/lnrpc._Lightning_SendPaymentSync_Handler.func1(0
x1763c80, 0xc002b48030, 0x1444520, 0xc000302240, 0x1, 0x1, 0xc000478940, 0xc0000
224b8)
github.com/lightningnetwork/lnd/lnrpc/rpc.pb.go:16082 +0x90
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1(0x1763c80,
0xc002b48030, 0x1444520, 0xc000302240, 0x20, 0x0, 0x0, 0xc0003e4601)
github.com/grpc-ecosystem/go-grpc-middleware@v1.0.0/chain.go:31 +0x127
github.com/lightningnetwork/lnd/rpcperms.(*InterceptorChain).MacaroonUnaryServer
Interceptor.func1(0x1763c80, 0xc002b48030, 0x1444520, 0xc000302240, 0xc00009e1c0
, 0xc00073c910, 0x0, 0x30001, 0x10000065a, 0x0)
github.com/lightningnetwork/lnd/rpcperms/interceptor.go:426 +0xcd
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1(0x1763c80,
0xc002b48030, 0x1444520, 0xc000302240, 0x0, 0xc001946b80, 0x400, 0x2030001)
github.com/grpc-ecosystem/go-grpc-middleware@v1.0.0/chain.go:34 +0xaf
github.com/lightningnetwork/lnd/rpcperms.(*InterceptorChain).rpcStateUnaryServer
Interceptor.func1(0x1763c80, 0xc002b48030, 0x1444520, 0xc000302240, 0xc00009e1c0
, 0xc00073c910, 0xc001f4d930, 0x40e769, 0xc00073c910, 0x50)
github.com/lightningnetwork/lnd/rpcperms/interceptor.go:512 +0xb3
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1(0x1763c80,
0xc002b48030, 0x1444520, 0xc000302240, 0x2e20f8, 0x50, 0xc00073c910, 0xde86e9)
github.com/grpc-ecosystem/go-grpc-middleware@v1.0.0/chain.go:34 +0xaf
github.com/lightningnetwork/lnd/rpcperms.errorLogUnaryServerInterceptor.func1(0x
1763c80, 0xc002b48030, 0x1444520, 0xc000302240, 0xc00009e1c0, 0xc00073c910, 0x1f
8b9a0, 0x0, 0xc001f4d9e0, 0x40eedf)
github.com/lightningnetwork/lnd/rpcperms/interceptor.go:342 +0x88
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1(0x1763c80, 0
xc002b48030, 0x1444520, 0xc000302240, 0xc00009e1c0, 0xc0000046d8, 0xc002b79a50,
0x5cd816, 0x139e300, 0xc002b48030)
github.com/grpc-ecosystem/go-grpc-middleware@v1.0.0/chain.go:39 +0x162
github.com/lightningnetwork/lnd/lnrpc._Lightning_SendPaymentSync_Handler(0x146dd
60, 0xc00007b860, 0x1763c80, 0xc002b48030, 0xc0028360c0, 0xc000506990, 0x1763c80
, 0xc002b48030, 0xc001636120, 0x8a)
github.com/lightningnetwork/lnd/lnrpc/rpc.pb.go:16084 +0x15e
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00000a160, 0x1771458, 0xc000
496300, 0xc002f44400, 0xc000506c30, 0x1f5f058, 0x0, 0x0, 0x0)
google.golang.org/grpc@v1.24.0/server.go:995 +0x482
google.golang.org/grpc.(*Server).handleStream(0xc00000a160, 0x1771458, 0xc000496
300, 0xc002f44400, 0x0)
google.golang.org/grpc@v1.24.0/server.go:1275 +0xd4c
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0005cca20, 0xc00000a160
, 0x1771458, 0xc000496300, 0xc002f44400)
google.golang.org/grpc@v1.24.0/server.go:710 +0xb2
created by google.golang.org/grpc.(*Server).serveStreams.func1
google.golang.org/grpc@v1.24.0/server.go:708 +0xa8
### Your environment

version: a329c8061266418bccd797aa5d7d277c736ee930

--- a/rpcserver.go
+++ b/rpcserver.go
@@ -3096,7 +3096,7 @@ func (r *rpcServer) PendingChannels(ctx context.Context,
 				return nil, err
 			}
 
-			resp.TotalLimboBalance += int64(forceClose.LimboBalance)
+			resp.TotalLimboBalance += forceClose.LimboBalance
 
 			resp.PendingForceClosingChannels = append(
 				resp.PendingForceClosingChannels,
@@ -4310,6 +4310,10 @@ func (r *rpcServer) extractPaymentIntent(rpcPayReq *rpcPaymentRequest) (rpcPayme
 	// Payment address may not be needed by legacy invoices.
 	if len(rpcPayReq.PaymentAddr) != 0 && len(rpcPayReq.PaymentAddr) != 32 {
 		return payIntent, errors.New("invalid payment address length")
+	}
+
+	if payIntent.paymentAddr == nil {
+		payIntent.paymentAddr = &[32]byte{}
 	}
 	copy(payIntent.paymentAddr[:], rpcPayReq.PaymentAddr)
 

